### PR TITLE
Update static dependency lint rule

### PR DESCRIPTION
**Pull request by kwridan**
Resolves https://github.com/tuist/tuist/issues/337

### Short description 📝

- The graph linter didn't account for which targets could link static products
- This led to raising an incorrect lint warning

### Solution 📦

Only flag a warning in the event a static product is linked twice (to targets that can link static products)

### Implementation 👩‍💻👨‍💻

- [x] Add a `canLinkStaticProducts` method to share link rules between the linter and graph linkable dependencies
- [x] Update fixture
- [x] Update changelog 

### Test Plan 🛠

- Run unit tests via `swift lint`
- Run acceptance tests via `bundle exec rake features`
- Manually generate `fixtures/ios_app_with_static_frameworks` via `tuist generate`
- Verify no lint warnings are displayed

### CODE REVIEW

File: Sources/TuistGenerator/Linter/GraphLinter.swift [hunk 1]
```diff
@@ -126,9 +114,27 @@ class GraphLinter: GraphLinting {
...
         return issues
     }
 
+    private func lintStaticDependencies(from: TargetNode, to: TargetNode) -> [LintingIssue] {
+        guard to.target.product.isStatic, from.target.canLinkStaticProducts() else {
+            return []
+        }
+        let warning = StaticDepedencyWarning(targetNode: from,
+                                             linkingStaticTargetNode: to)
+        let (inserted, oldMember) = linkedStaticProducts.insert(warning)
```
Comment (pepibumur) [inline Sources/TuistGenerator/Linter/GraphLinter.swift]: It was here before the PR, but I'd not hold state in the linter. If we call the linter several times with different projects that might result in invalid results.

File: Sources/TuistGenerator/Linter/GraphLinter.swift [hunk 2]
```diff
@@ -126,9 +114,27 @@ class GraphLinter: GraphLinting {
...
+        issues.append(contentsOf: lintStaticDependencies(from: from, to: to))
+
         return issues
     }
 
+    private func lintStaticDependencies(from: TargetNode, to: TargetNode) -> [LintingIssue] {
+        guard to.target.product.isStatic, from.target.canLinkStaticProducts() else {
+            return []
+        }
+        let warning = StaticDepedencyWarning(targetNode: from,
```
Comment (pepibumur) [inline Sources/TuistGenerator/Linter/GraphLinter.swift]: `targetNode` and `linkintStaticTargetNode` as names is misleading. I'd rather call them `fromTargetNode` and `toTargetNode`.

**Comment by kwridan** [comment]
Thanks, good suggestions 👍 !  I've applied them in the latest commit

**Comment by pepibumur** [comment]
Looks good to me! Great job @kwridan 

giphy (1)

**Comment by ollieatkinson** [comment]
Nice work @kwridan - changes look good to me.

**Pull request merged**